sir_model.py

```diff
@@ -18,17 +19,14 @@ def get_dummies(states):
 def get_infection_probas(states, transmissions):
     """
     - states[i] = state of i
-    - transmissions = array/list of i, j, lambda_ij
+    - transmissions = sparse matrix of i, j, lambda_ij
     - infection_probas[i]  = 1 - prod_{j: state==I} [1 - lambda_ij]
     """
     infected = (states == 1)
     N = len(states)
     infection_probas = np.zeros(N)
     for i in range(N):
-        rates = np.array([
-            rate for i0, j, rate in transmissions
-            if i0 == i and infected[j]
-        ])
+        rates = transmissions.toarray()[i][infected == True]
```

Comment:
same here we should try to replace the whole for loop by a sparse marix times vector

Reply:
My most recent commit has updated it by 
```python
def get_infection_probas(states, transmissions):
    """
    - states[i] = state of i
    - transmissions = sparse matrix of i, j, lambda_ij
    - infection_probas[i]  = 1 - prod_{j: state==I} [1 - lambda_ij]
    """
    infected = (states == 1)
    N = len(states)
    infected_transmissions = transmissions.multiply(infected)  # element-wise multiplication for masking
    infection_probas = (1 - (1-infected_transmissions.toarray()).prod(axis=1)).squeeze()
    assert len(infection_probas) == N  # sanity check
    return infection_probas
```